proposed/0006-Launcher.md

```diff
@@ -0,0 +1,192 @@
+```
+struct Response {
+    root_directory_id    : NameType,
+    root_directory_tag   : u64
+    public_signing_key   : sodiumoxide::crypto::sign::PublicKey,
+    private_signing_key  : sodiumoxide::crypto::sign::SecretKey,
+    public_encrytion_key : sodiumoxide::crypto::box_::PublicKey,
+    private_encrytion_key: sodiumoxide::crypto::box_::SecretKey,
+}
+```
+
+**step 8:** App does what it wants this point onwards.
+
+## Remove App Flow
+
+**procedure 0:** Launcher removes the App as follows:
+- Delete from `<LOCAL-CONFIG-FILE>` (on the user's machine) the following:
+```
+[
+    { App-ID, “/path/to/XYZ” }, // Remove this, other entries are untouched.
+    ...
+]
+```
+- Remove the SHA512(App-Binary) from the vector in `<LAUNCHER-CONFIG-FILE>`.
+- If the vector-size is **0** it means that this is the last machine where the App was present. In that case Launcher shall remove the App entry from the `<LAUNCHER-CONFIG-FILE>`. Launcher shall send a request to `MaidManagers` to un-map user's `MAID-Keys <-> App specific Keys`. The Launcher shall not delete `<APP-ROOT-DIR>` from within `SAFEDrive` folder. It is user's responsibility to do that as it might contain information (like pictures etc) which the user may not want to lose.
+
+**procedure 1:** While the other procedure would work, but there might be occassions when the user wants to immediately remove the app completely (which also translates as revoke App's permission to mutate network on user's behalf). He may not have access to other machines where the App was installed and maybe currently running and the previous procedure requires him to remove it from all machines to actually perform un-mapping of keys in `Vaults`. Thus there shall be an option in Launcher to remove App completely irrespective of if it is installed and/or running in other machines. In such cases Launcher will purge `Vec<SHA512(App-Binary)>` and proceed as above for the detection of empty vector.
+
+## Misc
+- If the App is added to Launcher in one machine, the mention of this will go into `<LAUNCHER-CONFIG-FILE>` as stated previously. It will thus be listed on every machine when user logs into his/her account via Launcher on that machine. However when the App is attempted to be activated on a machine via Launcher where it was not previously added to Launcher then he/she will be prompted to associate a binary. Once done, the information as usual will go into the `<LOCAL-CONFIG-FILE>` on that machine and the user won't be prompted the next time.
+- When the App is started via Launcher, it will first check if the `SHA512(App-Binary)` matches any one of the corresponding entries in the vector in `<LAUNCHER-CONFIG-FILE>`. If it does not Launcher will interpret this as a malacious App that has replaced the App-binary on user's machine, and thus will not pass any credentials to it nor communicate with it.
```

Comment:
It's more likely that the app has been updated here isn't it? I guess we should just seek user-confirmation of the change while warning them about the chance of malice?